ovos_PHAL_plugin_oauth/__init__.py

```diff
@@ -182,6 +188,47 @@ def handle_oauth_register(self, message):
             # eg. spotify
             self.oauth_skills[skill_id]["needs_creds"] = True
 
+    def handle_refresh_token(self, message):
+        """Refresh oauth token.
+
+        See:
+        https://www.oauth.com/oauth2-servers/making-authenticated-requests/refreshing-an-access-token/
+
+        for details on the procedure.
+        """
+        oauth_id = f"{message.data['skill_id']}_{message.data['app_id']}"
+        # Load all needed data for refresh
+        with OAuthApplicationDatabase() as db:
+            app_data = db[oauth_id]
+        with OAuthTokenDatabase() as db:
+            token_data = db[oauth_id]
+
+        if 'refresh_token' not in token_data:
+            LOG.warning("Token data doesn't contain a refresh token and cannot "
+                        "be refreshed.")
+            return
+
+        refresh_token = token_data['refresh_token']
+        refresh_endpoint = app_data["refresh_endpoint"]
+        client_id = app_data["client_id"]
+        client_secret = app_data["client_secret"]
+
+        # Perform refresh
+        client = WebApplicationClient(client_id, refresh_token=refresh_token)
+        uri, headers, body = client.prepare_refresh_token_request(refresh_endpoint)
+        refresh_result = requests.post(uri, headers=headers, data=body,
+                                       auth=(client_id, client_secret))
+
+        if refresh_result.ok:
+            new_token_data = refresh_result.json()
+            new_token_data['expires_at'] = time.time() + token_data['expires_in']
+            # Store token
+            with OAuthTokenDatabase() as db:
+                token_data.update(new_token_data)
+                db.update_token(oauth_id, token_data)
+        else:
+            LOG.error("Token refresh failed :(")
```

Comment:
Consider emitting a response message so the caller can be notified of success/failure

Reply:
Also to consider we already have a msg type returning the response from an initial fetch
https://github.com/OpenVoiceOS/ovos-PHAL-plugin-oauth/blob/126455896f0ff66c518404641299ce55887289d6/ovos_PHAL_plugin_oauth/__init__.py#L64-L66

Reply:
Is "oauth.token.response..." a good message name for the refresh result? Seems like it could be cause for confusion?